docs/root/intro/arch_overview/advanced/matching/matching_api.rst

```diff
@@ -148,6 +148,69 @@ with the tree matcher in an arbitrary order. The example describes the following
 the filter if ``some-header: skip_filter`` is present and ``second-header`` is set to *either* ``foo`` or
 ``bar``.
 
+Per-Route Configuration
+****************************
+
+The matcher can also be configured on a per-route basis.
+For example, to match only on a specific ``/static`` route:
+
+.. code-block:: yaml
+
+  route_config:
+    name: local_route
+    virtual_hosts:
+    - name: local_service
+      domains: ["*"]
+      typed_per_filter_config:
+        envoy.filters.http.match_delegate:
+          "@type": type.googleapis.com/envoy.extensions.common.matching.v3.ExtensionWithMatcherPerRoute
+          extension_config:
```

Comment:
this is not defined in the proto above. I suggest you nuke this example as its not complete.

Reply:
+1. We're going to get the extension_config overwrite in a followup right? The docs can be pushed back to that PR.

Reply:
hmm, `ExtensionWithMatcherPerRoute` and override are introduced and tested in this PR.

But i am ok with pushing back this example to follow-up PR.

Reply:
Then I'm a bit confused. The API changes only include the matcher override, not the extension config override.

Reply:
You are right. This example seems wrong. The per route config should be something like below.

```
typed_per_filter_config {
       key: filter_config_name
       value {
          [type.googleapis.com/envoy.extensions.common.matching.v3.ExtensionWithMatcherPerRoute] {
                xds_matcher {
                   ....
                }
```
@jstraceski , is this example up to date? can you share with me your full configuration? (via internal gPaste is good)

Reply:
@jstraceski  I think the example is out of date. I just checked the config you shared with me on May 15th. That one has correct format for per route config.

Anyway, please share with me your full config. Thanks.